GitCommands/ArgumentBuilderExtensions.cs

```diff
@@ -359,10 +369,23 @@ public static ArgumentString[] BuildBatchArguments(this ArgumentBuilder builder,
             // Add last commandline batch
             if (!lastBatchBuilder.IsEmpty)
             {
-                batches.Add(lastBatchBuilder.ToString());
+                batches.Add(new BatchArgumentItem(lastBatchBuilder, currentBatchItemCount));
             }
 
-            return batches.ToArray();
+            return batches;
+        }
+    }
+
+    public sealed class BatchArgumentItem
+    {
+        public BatchArgumentItem(ArgumentString argument, int count)
+        {
+            Argument = argument;
+            Count = count;
         }
+
+        public ArgumentString Argument { get; }
+
+        public int Count { get; }
```

Comment:
Please annotate with xml-docs these properties. I'm struggling to understand what each represents.